### PR TITLE
feat(spans): Scrub bools in SQL queries in span descriptions

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Scrub bools in span descriptions containing SQL queries. This is not only for the cardinality but also for a product requirement.

#skip-changelog

### CODE REVIEW

File: relay-general/src/store/regexes.rs [hunk 1]
```diff
@@ -68,7 +68,9 @@ pub static SQL_NORMALIZER_REGEX: Lazy<Regex> = Lazy::new(|| {
     ((?-x)(?P<single_quoted_strs>('(?:[^']|'')*?(?:\\'.*|[^']')))) |
     # Don't capture double-quoted strings (eg used for identifiers in PostgreSQL).
     # Capture numbers.
-    ((?-x)(?P<number>(-?\b(?:[0-9]+\.)?[0-9]+(?:[eE][+-]?[0-9]+)?\b)))
+    ((?-x)(?P<number>(-?\b(?:[0-9]+\.)?[0-9]+(?:[eE][+-]?[0-9]+)?\b))) |
+    # Capture booleans (as full tokens, not as substrings of other tokens).
+    ((?-x)(?P<bool>(\b(?:true|false)\b)))
```
Comment (olksdr) [inline relay-general/src/store/regexes.rs:73]: Looks good, but do we really have to scrub booleans?

**Review by olksdr — APPROVED** [review]

File: relay-general/src/store/regexes.rs (hunk 1 above)
Comment (jjbayer) [inline relay-general/src/store/regexes.rs:73]: Is this case sensitive? Because

`select tRuE or FaLsE` is valid postgreSQL.

Comment (jjbayer) [reply relay-general/src/store/regexes.rs:73]: +1, booleans are not high cardinality, so why bother?

**Review by jjbayer — APPROVED** [review]

File: relay-general/src/store/regexes.rs (hunk 1 above)
Comment (iker-barriocanal) [reply relay-general/src/store/regexes.rs:73]: Yes, the `i` modifier in the first line of the RegEx is for case insensitive. I added a test to ensure this in [a17ce05](https://github.com/getsentry/relay/pull/2121/commits/a17ce057c242072709f4b059a5bac555610ef936).

Comment (iker-barriocanal) [reply relay-general/src/store/regexes.rs:73]: In this case, scrubbing is not for high cardinality, but a product requirement. Updated the PR description.

**Pull request merged**